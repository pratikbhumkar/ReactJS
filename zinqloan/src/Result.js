import React, { Component } from "react";
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import App from './App';
import ReactDOM from 'react-dom';

class Result extends Component {

    constructor(props){
        super(props);
        if(typeof this.props.user !== 'undefined'){
          this.state={
            userObj:this.props.user,
          FirstName:this.props.user.FirstName,
          LastName:this.props.user.LastName
          }
        }
        else{
          var user={
            FirstName: 'First Name',
            LastName: 'Surname',
            status:'Couple',
            report:''
          }
          this.state={
          userObj:user,
          FirstName:'First Name',
          LastName:'Surname'
          }
        }
      }
      HandleLogout(event) {
        ReactDOM.render(<App />, document.getElementById('root'));
      }
    render() {
      return (
          <div>
                <MuiThemeProvider>
         <div >
         <form style={{textAlign:"center"}}>
         <AppBar title="Zinq" >
         <RaisedButton label="Log out" primary={true} style={{margin: 15}} onClick={(event) => this.HandleLogout(event)}/> 
         </AppBar>
      <h1> {this.state.FirstName} {this.state.LastName}</h1>
      <h1> {this.state.userObj.report} </h1>
      </form>
      </div>
          </MuiThemeProvider>
       </div>)
      }
    }
export default Result;