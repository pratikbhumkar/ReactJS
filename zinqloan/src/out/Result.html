<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Result</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Result</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Result<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">The result component displays whether the user is elible for the loan or not.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Result"><span class="type-signature"></span>new Result<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Result.js.html">Result.js</a>, <a href="Result.js.html#line12">line 12</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="HandleLogout"><span class="type-signature"></span>HandleLogout<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Logs out the user when clicking the Logout button.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Result.js.html">Result.js</a>, <a href="Result.js.html#line33">line 33</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Rendering the Result page.
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Result.js.html">Result.js</a>, <a href="Result.js.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Result.html">Result</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Wed Jan 16 2019 03:29:44 GMT+1100 (Australian Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>