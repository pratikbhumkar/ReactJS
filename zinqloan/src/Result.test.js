import React from 'react';
import Result from './Result';
import {shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  const editor =shallow(<Result/>) 
  expect(editor.find('AppBar').length).toEqual(1)
  expect(editor.find('h1').length).toEqual(2)
  expect(editor.find('RaisedButton').length).toEqual(1)
});
