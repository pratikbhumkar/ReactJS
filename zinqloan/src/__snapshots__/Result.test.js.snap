// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <form
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#00bcd4",
            "borderRadius": "0px",
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "flex",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "paddingLeft": 24,
            "paddingRight": 24,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
            "zIndex": 1100,
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": 48,
              "margin": 0,
              "marginLeft": -16,
              "marginRight": 8,
              "marginTop": 8,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": 12,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": 48,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "height": 24,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </div>
        </button>
        <h1
          onClick={[Function]}
          style={
            Object {
              "boxFlex": 1,
              "color": "#ffffff",
              "flex": "1",
              "fontSize": 24,
              "fontWeight": 400,
              "height": 64,
              "letterSpacing": 0,
              "lineHeight": "64px",
              "margin": 0,
              "muiPrepared": true,
              "overflow": "hidden",
              "paddingTop": 0,
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Zinq
        </h1>
        <div
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "margin": 15,
              "minWidth": 88,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            id="btnLogout"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#00bcd4",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Log out
                </span>
              </div>
            </div>
          </button>
        </div>
      </div>
      <h1>
         
        FirstName
         
        LastName
      </h1>
      <h1>
         
        
         
      </h1>
    </form>
  </div>
</div>
`;
